src/compass/s1_geo_stack.py

```diff
@@ -0,0 +1,566 @@
+    x_spac: float
+        Spacing of geocoded burst along X-direction
+    y_spac: float
+        Spacing of geocoded burst along Y-direction
+    epsg: int
+        EPSG code identifying projection system to use for processing
+    flatten: bool
+        Enable/disable flattening of geocoded burst
+    is_split_spectrum: bool
+        Enable/disable range split spectrum
+    low_band: float
+        Low sub-band bandwidth for split-spectrum in Hz
+    high_band: float
+        High sub-band bandwidth for split-spectrum in Hz
+    '''
+    start_time = time.time()
+    error = journal.error('s1_geo_stack_processor.main')
+    info = journal.info('s1_geo_stack_processor.main')
+
+    # Check if SLC dir and DEM exists
+    if not os.path.isdir(slc_dir):
+        err_str = f'{slc_dir} SLC directory does not exist'
+        error.log(err_str)
+        raise FileNotFoundError(err_str)
+
+    if not os.path.isfile(dem_file):
+        err_str = f'{dem_file} DEM file does not exists'
+        error.log(err_str)
+        raise FileNotFoundError(err_str)
+
+    # Create directory for runfiles
+    run_dir = f'{work_dir}/run_files'
+    os.makedirs(run_dir, exist_ok=True)
+
+    # Check if orbit are provided, if Not download
+    if orbit_dir is None:
+        info.log('Orbit directory not assigned. Download orbit ephemerides')
+
+        # Create orbit dir and check internet connection
+        orbit_dir = f'{work_dir}/orbits'
+        os.makedirs(orbit_dir, exist_ok=True)
+        check_internet_connection()
+
+        # List all zip file and extract info
+        zip_file_list = sorted(glob.glob(f'{slc_dir}/S1*zip'))
+
+        for zip_file in zip_file_list:
+            sensor_id, _, start_datetime, \
+            end_datetime, _ = parse_safe_filename(zip_file)
+
+            # Find precise orbits first
+            orbit_dict = get_orbit_dict(sensor_id, start_datetime,
+                                        end_datetime, 'AUX_POEORB')
+            # If orbit_dict is empty, precise orbits have not been found. Find restituted orbits instead
+            if orbit_dict == None:
+                orbit_dict = get_orbit_dict(sensor_id, start_datetime,
+                                            end_datetime, 'AUX_RESORB')
+            # Download the orbit file
+            download_orbit(orbit_dir, orbit_dict['orbit_url'])
+
+    # Generate burst map and prune it if a list of burst ID is provided
+    zip_file = sorted(glob.glob(f'{slc_dir}/S1*zip'))
+    burst_map = generate_burst_map(zip_file, orbit_dir, x_spac, y_spac, epsg)
+
+    # Identify burst IDs common across the stack and remove from the dataframe
+    # burst IDs that are not in common
+    common_ids = get_common_burst_ids(burst_map)
+    burst_map = prune_dataframe(burst_map, 'burst_id', common_ids)
+
+    # If user selects burst IDs to process, prune unnecessary bursts
+    if burst_id is not None:
+        burst_map = prune_dataframe(burst_map, 'burst_id', burst_id)
```

Comment:
Can avoid this `prune_dataframe` call if  `generate_burst_map` takes in `burst_id` and uses it to filter bursts. Like [this](https://github.com/opera-adt/COMPASS/pull/14#discussion_r915167917).